codec-http/src/main/java/io/netty/handler/codec/http/multipart/AbstractDiskHttpData.java

```diff
@@ -165,9 +164,7 @@ public void addContent(ByteBuf buffer, boolean last)
                     RandomAccessFile accessFile = new RandomAccessFile(file, "rw");
                     fileChannel = accessFile.getChannel();
                 }
-                while (written < localsize) {
-                    written += fileChannel.write(byteBuffer);
-                }
+                buffer.getBytes(buffer.readerIndex(), fileChannel, fileChannel.position(), localsize);
```

Comment:
@normanmaurer The javadocs for `ByteBuf#getBytes' state that the `fileChannel` position is not being modified. Does this not imply that all calls to this method start writing from the beginning of the file?

`written` is now also effectively unused; shouldn't this line be `written = buffer.getBytes(...);`?.